### PR TITLE
feat: postgres query stream

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #5944 

### Description:

Captura de ecrã 2023-06-19, às 14 39 35

### CODE REVIEW

File: packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts [hunk 1]
```diff
@@ -138,32 +140,86 @@ export class PostgresClient<
...
+            });
+        })
+            .catch((e) => {
+                throw new WarehouseQueryError(
+                    `Error running postgres query: ${e}`,
+                );
+            })
+            .finally(() => {
+                pool?.end().catch(() => {
+                    console.info('Failed to end postgres pool');
```
Comment (rephus) [inline packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts:220]: shoudl this be a Logger.error ? + captureSentry ?

File: packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts [hunk 2]
```diff
@@ -138,32 +140,86 @@ export class PostgresClient<
...
+                            rows.push(chunk.row);
+                            fields = chunk.fields;
+                            callback();
+                        },
+                    }),
+                );
+            });
+        })
+            .catch((e) => {
+                throw new WarehouseQueryError(
```
Comment (rephus) [inline packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts:214]: perhaps we should add a captureSentry to monitor these errors

File: packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts [hunk 3]
```diff
@@ -138,32 +140,86 @@ export class PostgresClient<
         this.config = config;
     }
 
+    private getSQLWithMetadata(sql: string, tags?: Record<string, string>) {
+        let alteredQuery = sql;
```
Comment (rephus) [inline packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts:144]: How about 

```
if (tags) return`${sql}\n-- ${JSON.stringify(tags)}`
else return sql 


```

File: packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts [hunk 4]
```diff
@@ -138,32 +140,86 @@ export class PostgresClient<
...
+        if (tags) {
+            alteredQuery = `${alteredQuery}\n-- ${JSON.stringify(tags)}`;
```
Comment (rephus) [inline packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts:146]: I'm not sure what this is doing

**Review by rephus — COMMENTED** [review]
Code seems ok, I thought you had a problem with sql runner and getting metadata, did you find  a way to solve this ?

**Review by rephus — APPROVED** [review]
I tested this on render, everything seems fine, including sql runner

File: packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts (hunk 1 above)
Comment (ZeRego) [reply packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts:220]: Warehouse package doesn't have access to those at the moment.

File: packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts (hunk 4 above)
Comment (ZeRego) [reply packages/warehouses/src/warehouseClients/PostgresWarehouseClient.ts:146]: This logic was introduced in this [PR](https://github.com/lightdash/lightdash/pull/5784), I just moved it to a private function.

**Pull request merged**